azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/service/impl/FunctionAppDeploymentSlot.java

```diff
@@ -31,11 +33,23 @@
 
 public class FunctionAppDeploymentSlot extends FunctionAppBase<FunctionDeploymentSlot, FunctionAppDeploymentSlotEntity>
         implements IFunctionAppDeploymentSlot {
-
+    private static final String FUNCTION_DEPLOYMENT_SLOT_ID_TEMPLATE = "/subscriptions/%s/resourceGroups/%s/providers/Microsoft.Web/sites/%s/slots/%s";
+    @Nonnull
+    private final String functionAppName;
     private final AzureResourceManager azureClient;
 
-    public FunctionAppDeploymentSlot(FunctionAppDeploymentSlotEntity deploymentSlot, AzureResourceManager azureClient) {
-        this.entity = deploymentSlot;
+    private FunctionApp parent;
```

Comment:
every `FunctionAppDeployment` must belongs to a `FunctionApp`
so make it `final`